api/base/serializers.py

```diff
@@ -280,6 +299,54 @@ def to_representation(self, data):
             self.child.to_representation(item, envelope=None) for item in data
         ]
 
+    # Overrides ListSerializer which doesn't support multiple update by default
+    def update(self, instance, validated_data):
+        if len(instance) != len(validated_data):
+            raise exceptions.NotFound()
```

Comment:
I'm not sure `404 Not Found` is the correct status. 404s are returned when requesting a resource that doesn't exist. In this case, the client is sending an invalid payload. They might even be sending an array that's too long--it could contain  existing, valid resource objects along with non-existent or invalid resource objects, in which case 404 would be inappropriate.

I think the rest_framework_bulk package you linked to is correct in raising a `ValidationError`.